luxtronik/__init__.py

```diff
@@ -103,41 +107,83 @@ def _with_lock_and_connect(self, func, *args, **kwargs):
             ret_val = func(*args, **kwargs)
             return ret_val
 
-    def read(self):
+    def read(self, data=None):
         """
-        Read data from heat pump.
-        All available data will be read from the heat pump.
+        All available data will be read from the heat pump
+        and integrated to the passed data object.
+        This data object is returned afterwards, mainly for access to a newly created.
         """
-        return self._with_lock_and_connect(self._read)
+        if data is None:
+            data = LuxtronikData()
+        return self._with_lock_and_connect(self._read, data)
 
-    def read_parameters(self):
-        """Read parameters from heat pump."""
-        return self._with_lock_and_connect(self._read_parameters)
+    def read_parameters(self, parameters=None):
+        """
+        Read parameters from heat pump and integrate them to the passed dictionary.
+        This dictionary is returned afterwards, mainly for access to a newly created.
+        """
+        if parameters is None:
+            parameters = Parameters()
+        return self._with_lock_and_connect(self._read_parameters, parameters)
 
-    def read_calculations(self):
-        """Read calculations from heat pump."""
-        return self._with_lock_and_connect(self._read_calculations)
+    def read_calculations(self, calculations=None):
+        """
+        Read calculations from heat pump and integrate them to the passed dictionary.
+        This dictionary is returned afterwards, mainly for access to a newly created.
+        """
+        if calculations is None:
+            calculations = Calculations()
+        return self._with_lock_and_connect(self._read_calculations, calculations)
 
-    def read_visibilities(self):
-        """Read visibilities from heat pump."""
-        return self._with_lock_and_connect(self._read_visibilities)
+    def read_visibilities(self, visibilities=None):
+        """
+        Read visibilities from heat pump and integrate them to the passed dictionary.
+        This dictionary is returned afterwards, mainly for access to a newly created.
+        """
+        if visibilities is None:
+            visibilities = Visibilities()
+        return self._with_lock_and_connect(self._read_visibilities, visibilities)
 
     def write(self, parameters):
         """
-        Write parameter to heat pump.
-        All parameters will be written to the heat pump
-        prior to reading back in all data from the heat pump.
+        Write all set parameters to the heat pump.
         :param Parameters() parameters  Parameter dictionary to be written
                           to the heatpump before reading all available data
                           from the heat pump.
         """
-        return self._with_lock_and_connect(self._write, parameters)
+        self._with_lock_and_connect(self._write, parameters)
+
+    def write_parameter(self, target, value, safe=True):
```

Comment:
Do we really need this new function?

Reply:
In my opinion it is a good addition. Similar to `write_and_read()` where you could also call `write()` and then `read()`, here it is with `parameters.set()` and `write()`.

By the way, the `queue` should actually be independent of the `Parameters` object. But that is not part of this pull request. It would just be a start.

Reply:
I am not sure if it is a good idea. In any case, this should not go into this PR.

Reply:
Removed